[Bug]: Possible missing file close in esstracore.cpp::collect_paths

### Mandatory confirmtion

- [x] follow this project's [Code of Conduct](https://github.com/sony/esstra/blob/main/CODE_OF_CONDUCT.md).
- [x] follow this project's [Contributing Guidelines](https://github.com/sony/esstra/blob/main/CONTRIBUTING.md).
- [x] have checked the [issue list](https://github.com/sony/esstra/issues), and confirm that a similar issue has not been reported before.

### Description

When trying to compile the kernel, with esstra support enabled, I encountered the following error
`/home/tbird/work/ESSTRA/linux/security/selinux/hooks.c:97:10: fatal error: /home/tbird/work/ESSTRA/linux/security/selinux/avc.h: Too many open files
compilation terminated.`

I checked my open files limit (using ulimit) and found it was 1024.  I don't get this problem when  compiling
without esstra.  If I change my file open limit to 4096, the error goes away.  (e.g. 'ulimit -n 4096')

In examining the code in esstracore.cpp::collect_paths, I see an open, but no corresponding close.  I think the code may be leaving the file open every time collect_paths() is called.


### Version of ESSTRA

v0.4.0-8-g0be650f

### Environment

I'm building on Ubuntu 24.04, using gcc version: gcc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0


### Steps to reproduce

In the esstra directory:
$ sudo make install
$ sudo make install-spec
In the kernel source directory:
$ ttc timdesk setenv
$ ttc get_config
$ ttc set_config CONFIG_ESSTRA_SECTION=y
$ ttc kbuild (which results in 'make -j12 bzImage')


### Console logs, visual proof, etc.

_No response_

### Expected results

_No response_

### Additional context

_No response_

--- a/core/esstracore.cpp
+++ b/core/esstracore.cpp
@@ -225,28 +225,28 @@ collect_paths(void* gcc_data, void* /* user_data */) {
 
     uint8_t* buffer = new uint8_t[st_size];
     ssize_t size = read(fd, buffer, st_size);
+    close(fd);
+
     if (size != st_size) {
         fprintf(stderr, "size mismatch: st_size:%lu, read():%zd\n", st_size, size);
-        return;
-    }
+    } else {
+      FileInfo finfo;
+      // calculate hashes with specified algorithms
+      finfo[key_sha1] = "'" + calc_sha1(buffer, size) + "'";
 
-    // calculate hashes with specified algorithms
-    FileInfo finfo;
-    finfo[key_sha1] = "'" + calc_sha1(buffer, size) + "'";
-
-    // just for demonstration
-    for (const auto &algo: specified_algos) {
+      // just for demonstration
+      for (const auto &algo: specified_algos) {
         debug("calculate '%s' hash", algo.c_str());
         if (algo == "md5") {
-            finfo[key_md5] = "'" + calc_md5(buffer, size) + "'";
+          finfo[key_md5] = "'" + calc_md5(buffer, size) + "'";
         } else if (algo == "sha256") {
-            finfo[key_sha256] = "'" + calc_sha256(buffer, size) + "'";
+          finfo[key_sha256] = "'" + calc_sha256(buffer, size) + "'";
         } else {
-            fprintf(stderr, "unsupported hash algorithm '%s'\n", algo.c_str());
+          fprintf(stderr, "unsupported hash algorithm '%s'\n", algo.c_str());
         }
+      }
+      infomap[resolved] = finfo;
     }
-
-    infomap[resolved] = finfo;
 
     delete[] buffer;
 }

